src/TrackArtist.cpp

```diff
@@ -260,6 +260,68 @@ void TrackArtist::UpdatePrefs()
    SetColours(0);
 }
 
+void TrackArt::DrawClipAffordance(wxDC& dc, const wxRect& rect, bool highlight, bool selected, const wxString& title)
+{
+    // bottom rounded corners supposed to be hidden using clipping region or overdraw
+   if (selected)
+   {
+      wxRect strokeRect{
+         rect.x - ClipSelectionStrokeSize,
+         rect.y,
+         rect.width + ClipSelectionStrokeSize * 2,
+         rect.height + ClipFrameRadius };
+      dc.SetBrush(*wxTRANSPARENT_BRUSH);
+      AColor::UseThemeColour(&dc, clrClipAffordanceStroke, clrClipAffordanceStroke);
+      dc.DrawRoundedRectangle(strokeRect, ClipFrameRadius);
+   }
+   AColor::UseThemeColour(&dc, highlight ? clrClipAffordanceActiveBrush : clrClipAffordanceInactiveBrush, clrClipAffordanceOutlinePen);
+   dc.DrawRoundedRectangle(wxRect(rect.x, rect.y + ClipSelectionStrokeSize, rect.width, rect.height + ClipFrameRadius), ClipFrameRadius);
+   if (!title.empty())
+   {
+      dc.SetFont(wxFont(8, wxFONTFAMILY_SWISS, wxFONTSTYLE_NORMAL, wxFONTWEIGHT_NORMAL));
+      dc.SetTextForeground(*wxBLACK);
+      auto ascent = dc.GetFontMetrics().ascent;
+      dc.DrawText(title, rect.x + ClipFrameRadius, rect.y + (rect.height + ascent) / 2 - ascent);
+   } 
+}
+
+void TrackArt::DrawClipEdges(wxDC& dc, const wxRect& clipRect, bool selected)
+{
+   dc.SetBrush(*wxTRANSPARENT_BRUSH);
+   {
+      AColor::UseThemeColour(&dc, -1, clrClipAffordanceOutlinePen);
+      AColor::Line(dc,
+         clipRect.GetLeft(), clipRect.GetTop(),
+         clipRect.GetLeft(), clipRect.GetBottom());
+      AColor::Line(dc,
+         clipRect.GetRight(), clipRect.GetTop(),
+         clipRect.GetRight(), clipRect.GetBottom());
+   }
+   if(selected)
+   {
+      if constexpr (ClipSelectionStrokeSize == 1)
+      {
+         AColor::UseThemeColour(&dc, -1, clrClipAffordanceStroke);
+         AColor::Line(dc,
+            clipRect.GetLeft() - ClipSelectionStrokeSize, clipRect.GetTop(),
+            clipRect.GetLeft() - ClipSelectionStrokeSize, clipRect.GetBottom());
+         AColor::Line(dc,
+            clipRect.GetRight() + ClipSelectionStrokeSize, clipRect.GetTop(),
+            clipRect.GetRight() + ClipSelectionStrokeSize, clipRect.GetBottom());
+      }
+      else if constexpr (ClipSelectionStrokeSize > 1)
+      {
+         AColor::UseThemeColour(&dc, clrClipAffordanceStroke, clrClipAffordanceStroke);
+         dc.DrawRectangle(wxRect(
+            clipRect.GetLeft() - ClipSelectionStrokeSize, clipRect.GetTop(),
+            ClipSelectionStrokeSize, clipRect.GetHeight()));
+         dc.DrawRectangle(wxRect(
+            clipRect.GetRight() + 1, clipRect.GetTop(),
+            ClipSelectionStrokeSize, clipRect.GetHeight()));
+      }
+   }
+}
+
```

Comment:
The TrackArtists additions might make a small separate commit, followed by commits for wave and note tracks